Tests/OpenCombineTests/PublisherTests/CountTests.swift

```diff
@@ -0,0 +1,169 @@
+//
+//  CountTests.swift
+//
+//
+//  Created by Joseph Spadafora on 6/25/19.
+//
+
+import XCTest
+
+#if OPENCOMBINE_COMPATIBILITY_TEST
+import Combine
+#else
+import OpenCombine
+#endif
+
+@available(macOS 10.15, *)
+final class CountTests: XCTestCase {
+
+    static let allTests = [
+        ("testSendsCorrectCount", testSendsCorrectCount),
+        ("testCountWaitsUntilFinishedToSend", testCountWaitsUntilFinishedToSend),
+        ("testDemand", testDemand),
+        ("testNoDemand", testNoDemand),
+        ("testAddingSubscriberRequestsUnlimitedDemand",
+            testAddingSubscriberRequestsUnlimitedDemand)
+    ]
+
+    func testSendsCorrectCount() {
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let countPublisher = publisher.count()
+        let tracking = TrackingSubscriber(
+            receiveSubscription: { $0.request(.max(42)) }
+        )
+
+        XCTAssertEqual(tracking.history, [])
+
+        countPublisher.subscribe(tracking)
+        XCTAssertEqual(tracking.history, [.subscription("Count")])
+
+        let sendAmount = Int.random(in: 1...1000)
+        for _ in 0..<sendAmount {
+            _ = publisher.send(3)
+        }
+        XCTAssertEqual(tracking.history, [.subscription("Count")])
+
+        publisher.send(completion: .finished)
+        XCTAssertEqual(tracking.history, [.subscription("Count"),
+                                          .value(sendAmount),
+                                          .completion(.finished)])
+    }
+
+    func testCountWaitsUntilFinishedToSend() {
```

Comment:
Isn't this test equivalent to the test above?

Reply:
The both cover some of the same things, but the one above is focused on checking specifically that the count is correct.  The bottom test is checking that the history doesn't have .value elements added after each send event.